### PR TITLE
Add a note about --dry-run displaying secrets

### PR DESCRIPTION
**Pull request by mattfarina**
**What this PR does / why we need it**: It adds a message to the install and upgrade help output about `--dry-run` and what it outputs. That includes secrets.

**Special notes for your reviewer**: N/A

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/install.go [hunk 1]
```diff
@@ -94,7 +94,10 @@ And in the following example, 'foo' is set to '{"key1":"value1","key2":"bar"}':
     $ helm install --set-json='foo={"key1":"value1","key2":"value2"}' --set-json='foo.key2="bar"' myredis ./redis
 
 To check the generated manifests of a release without installing the chart,
-the '--debug' and '--dry-run' flags can be combined.
+the --debug and --dry-run flags can be combined.
+
+The --dry-run flag will output all generated chart manifests, including Secrets.
```
Comment (sabre1041) [inline cmd/helm/install.go:99]:
```suggestion
The --dry-run flag will output all generated chart manifests, including Secrets which can be decoded exposing the underlying values.
```

File: cmd/helm/install.go [hunk 2]
```diff
@@ -94,7 +94,10 @@ And in the following example, 'foo' is set to '{"key1":"value1","key2":"bar"}':
...
+Please carefully consider the environment it is used in.
```
Comment (sabre1041) [inline cmd/helm/install.go:100]:
```suggestion
Please carefully consider how and when these arguments are used.
```

**Review by sabre1041 — COMMENTED** [review]
Small suggestion to refine the verbiage. If approved, please be sure to update the wording in both `install.go` and `upgrade.go` files

File: cmd/helm/install.go (hunk 1 above)
Comment (mattfarina) [reply cmd/helm/install.go:99]: Because these secrets are chart/template generated they may be plain text. This is before they are sent to Kubernetes.

Comment (sabre1041) [reply cmd/helm/install.go:99]: Update Updated proposed changes

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by mattfarina** [comment]
Note, this documentation is converted into mardown and placed on helm.sh.

**Pull request merged**